presto-common/src/main/java/com/facebook/presto/common/zorder/ZOrder.java

```diff
@@ -0,0 +1,371 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.zorder;
+
+import com.facebook.presto.common.type.Type;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+public final class ZOrder
+{
+    private ZOrder()
+    {
+    }
+
+    private static final int binaryIntStringLength = 32;
+
+    private static String repeatString(String string, Integer repeat)
+    {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < repeat; i++) {
+            stringBuilder.append(string);
+        }
+        return stringBuilder.toString();
+    }
+
+    private static int flipBits(int n)
+    {
+        return ~n;
+    }
+
+    private static String flipBits(String binaryString)
+    {
+        char[] charArray = binaryString.toCharArray();
+        for (int i = 0; i < charArray.length; i++) {
+            charArray[i] = (charArray[i] == '1') ? '0' : '1';
+        }
+        return String.valueOf(charArray);
+    }
+
+    private static String flipFirstBit(String binaryString)
+    {
+        char[] charArray = binaryString.toCharArray();
+        charArray[0] = (charArray[0] == '1') ? '0' : '1';
+        return String.valueOf(charArray);
+    }
+
+    public static int maxBitSize(int value)
+    {
+        int count = 0;
+        while (value != 0) {
+            value /= 2;
+            count++;
+        }
+        return count;
+    }
+
+    public static int maxBitSize(long value)
+    {
+        int count = 0;
+        while (value != 0) {
+            value /= 2;
+            count++;
+        }
+        return count;
+    }
+
+    public static long encodeIntegers(List<Integer> cols)
+    {
+        int colSize = cols.size();
+        int maxValue = 0;
+        for (int col = 0; col < colSize; col++) {
+            int column = cols.get(col);
+            if (column < 0) {
+                column = -(column + 1);
+            }
+            maxValue = Math.max(maxValue, column);
+        }
+
+        int maxBitSize = maxBitSize(maxValue);
+        int bitPosition = 0;
+        long address = 0;
+        for (int bitPos = 0; bitPos < maxBitSize; bitPos++) {
+            for (int col = 0; col < colSize; col++) {
+                int column = cols.get(col);
+                if (column < 0) {
+                    column = -(column + 1);
+                }
+                address |= (long) ((column >> bitPos) & 1) << bitPosition++;
+            }
+        }
+        for (int col = colSize - 1; col >= 0; col--) {
+            int column = cols.get(col);
+            if (column >= 0) {
+                address |= (long) 1 << bitPosition;
+            }
+            bitPosition++;
```

Comment:
Can you explain a bit in a comment? I know this tries to keep the order of the numbers. One question for me is: by doing it, can we make it continuous addresses?